src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java

```diff
@@ -0,0 +1,94 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.imports.RedundantImportCheck;
+
+public class XpathRegressionRedundantImportTest extends AbstractXpathTestSupport {
+
+    private final String checkName = RedundantImportCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne(List<String> expectedXpathQueries) throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionRedundantImport1.java"));
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(RedundantImportCheck.class);
+        final String[] expectedViolation = {
+                "3:104: " + getCheckMessage(RedundantImportCheck.class,
+                        RedundantImportCheck.MSG_KEY, "org.checkstyle.suppressionxpathfilter.redundantimport.SuppressionXpathRegressionRedundantImport1"),
+        };
+        final List<String> expectedXPathQueries = Collections.singletonList(
+                "/IMPORT/DOT[./IDENT[@text='SuppressionXpathRegressionRedundantImport1']]/DOT[./IDENT[@text='redundantimport']]"
+                        + "/DOT[./IDENT[@text='suppressionxpathfilter']]/DOT[./IDENT[@text='checkstyle']]"
+                        + "/DOT/IDENT[@text='org']");
```

Comment:
This Xpath query isn't right, it should be `/COMPILATION_UNIT/IMPORT`

Reply:
Done

Reply:
Resolved